src/main/java/org/springframework/data/redis/connection/lettuce/LettuceReactiveRedisConnection.java

```diff
@@ -51,6 +51,8 @@ class LettuceReactiveRedisConnection implements ReactiveRedisConnection {
 	private final AsyncConnect<StatefulConnection<ByteBuffer, ByteBuffer>> dedicatedConnection;
 	private final AsyncConnect<StatefulRedisPubSubConnection<ByteBuffer, ByteBuffer>> pubSubConnection;
 
+	private final LettuceReactivePubSubCommands pubSub = new LettuceReactivePubSubCommands(this);
```

Comment:
Does it make sense to have this one as `Lazy<LettuceReactivePubSubCommands>`?

Reply:
`Lazy` isn't synchronized. On the other hand, creating concurrent hash maps isn't ideal either. Maybe a `synchronized` access with lazy creation in the getter is a better way out.